tests/trace/test_dataset.py

```diff
@@ -36,6 +37,81 @@ def test_pythonic_access(client):
         ds[-1]
 
 
+def _top_level_logs(log):
+    """Strip out internal logs from the log list"""
+    return [l for l in log if not l.startswith("_")]
+
+
+def test_dataset_laziness(client):
+    """
+    The intention of this test is to show that local construction of
+    a dataset does not trigger any remote operations.
+    """
+    dataset = Dataset(rows=[{"input": i} for i in range(300)])
+    log = client.server.attribute_access_log
+    assert _top_level_logs(log) == ["ensure_project_exists"]
```

Comment:
aside: interesting that we do this on Dataset init and not on save

Reply:
yeah fair point, why do we need that validation/interaction when its just in mem? probably confusing to the user